masonite/providers/RouteProvider.py

```diff
@@ -127,13 +125,42 @@ def boot(self, route_class: Route, request_class: Request):
                 | This is middleware with an after method.
                 |
                 """
-    
+
+                # Loads the request in so the middleware
+                # specified is able to use the
+                # request object.
+                route.run_middleware('after')
+
                 for http_middleware in self.app.make('HttpMiddleware'):
                     located_middleware = self.app.resolve(
                         http_middleware
                     )
                     if hasattr(located_middleware, 'after'):
                         located_middleware.after()
+
+                # Breaks the loop because the incoming route is found and executed.
+                # There is no need to continue searching WebRoutes.
                 break
             else:
                 self.app.bind('Response', 'Route not found. Error 404')
+
+    def create_matchurl(self, router, route):
```

Comment:
let's put this in the helpers inside `masonite/helpers` directory. you can put it in the `routes.py` file